### PR TITLE
En 8336/rename auction sc

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: cmd/node/config/systemSmartContractsConfig.toml [hunk 1]
```diff
@@ -10,7 +10,6 @@
     MaximumPercentageToBleed = 0.5
     BleedPercentagePerRound = 0.00001
     MaxNumberOfNodesForStake = 36
-    NodesToSelectInAuction = 36
```
Comment (iulianpascalau) [inline cmd/node/config/systemSmartContractsConfig.toml:13]: :+1:

File: vm/systemSmartContracts/auction.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/ElrondNetwork/protobuf/protobuf  --gogoslick_out=. auction.proto
+//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/ElrondNetwork/protobuf/protobuf  --gogoslick_out=. validator.proto
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: could have renamed also the auction.go file in a separate commit

File: vm/systemSmartContracts/auction.go [hunk 3]
```diff
@@ -92,8 +88,7 @@ func NewStakingAuctionSmartContract(
 		return nil, vm.ErrNilEpochNotifier
 	}
 
-	baseConfig := AuctionConfig{
-		NumNodes:    uint32(args.NumOfNodesToSelect),
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go:96]: The removal of the NumNodes config value will break the backwards compatibility?

File: vm/systemSmartContracts/auction.go [hunk 4]
```diff
@@ -626,78 +616,78 @@ func (s *stakingAuctionSC) registerBLSKeys(
...
 
 	registrationData.TotalStakeValue.Add(registrationData.TotalStakeValue, args.CallValue)
-	if registrationData.TotalStakeValue.Cmp(auctionConfig.NodePrice) < 0 &&
+	if registrationData.TotalStakeValue.Cmp(validatorConfig.NodePrice) < 0 &&
 		!core.IsSmartContractAddress(args.CallerAddr) {
-		s.eei.AddReturnMessage(
-			fmt.Sprintf("insufficient stake value: expected %s, got %s",
-				auctionConfig.NodePrice.String(),
+		v.eei.AddReturnMessage(
+			fmt.Sprintf("insufficient stake value: expected %v, got %v",
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: Please revert here

File: vm/systemSmartContracts/auction.go [hunk 5]
```diff
@@ -801,28 +791,28 @@ func (s *stakingAuctionSC) activateStakingFor(
...
+		vmOutput, err := v.executeOnStakingSC([]byte("stake@" +
 			hex.EncodeToString(currentBLSKey) + "@" +
 			hex.EncodeToString(rewardAddress) + "@" +
 			hex.EncodeToString(ownerAddress),
 		))
 		if err != nil {
-			s.eei.AddReturnMessage(fmt.Sprintf("cannot do stake for key %s, error %s", hex.EncodeToString(currentBLSKey), err.Error()))
-			s.eei.Finish(currentBLSKey)
-			s.eei.Finish([]byte{failed})
+			v.eei.AddReturnMessage(fmt.Sprintf("cannot do stake for key %v, error %v", hex.EncodeToString(currentBLSKey), err.Error()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 6]
```diff
@@ -801,28 +791,28 @@ func (s *stakingAuctionSC) activateStakingFor(
...
+			v.eei.Finish(currentBLSKey)
+			v.eei.Finish([]byte{failed})
 			continue
 
 		}
 		if vmOutput.ReturnCode != vmcommon.Ok {
-			s.eei.AddReturnMessage(fmt.Sprintf("cannot do stake for key %s, error %s", hex.EncodeToString(currentBLSKey), vmOutput.ReturnCode.String()))
-			s.eei.Finish(currentBLSKey)
-			s.eei.Finish([]byte{failed})
+			v.eei.AddReturnMessage(fmt.Sprintf("cannot do stake for key %v, error %v", hex.EncodeToString(currentBLSKey), vmOutput.ReturnCode.String()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 7]
```diff
@@ -834,185 +824,185 @@ func (s *stakingAuctionSC) activateStakingFor(
...
 
-func (s *stakingAuctionSC) setOwnerOfBlsKey(blsKey []byte, ownerAddress []byte) bool {
-	vmOutput, err := s.executeOnStakingSC([]byte("setOwner@" + hex.EncodeToString(blsKey) + "@" + hex.EncodeToString(ownerAddress)))
+func (v *validatorSC) setOwnerOfBlsKey(blsKey []byte, ownerAddress []byte) bool {
+	vmOutput, err := v.executeOnStakingSC([]byte("setOwner@" + hex.EncodeToString(blsKey) + "@" + hex.EncodeToString(ownerAddress)))
 	if err != nil {
-		s.eei.AddReturnMessage(fmt.Sprintf("cannot set owner for key %s, error %s", hex.EncodeToString(blsKey), err.Error()))
-		s.eei.Finish(blsKey)
-		s.eei.Finish([]byte{failed})
+		v.eei.AddReturnMessage(fmt.Sprintf("cannot set owner for key %v, error %v", hex.EncodeToString(blsKey), err.Error()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 8]
```diff
@@ -834,185 +824,185 @@ func (s *stakingAuctionSC) activateStakingFor(
...
+		v.eei.Finish(blsKey)
+		v.eei.Finish([]byte{failed})
 		return false
 
 	}
 	if vmOutput.ReturnCode != vmcommon.Ok {
-		s.eei.AddReturnMessage(fmt.Sprintf("cannot set owner for key %s, error %s", hex.EncodeToString(blsKey), vmOutput.ReturnCode.String()))
-		s.eei.Finish(blsKey)
-		s.eei.Finish([]byte{failed})
+		v.eei.AddReturnMessage(fmt.Sprintf("cannot set owner for key %v, error %v", hex.EncodeToString(blsKey), vmOutput.ReturnCode.String()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 9]
```diff
@@ -834,185 +824,185 @@ func (s *stakingAuctionSC) activateStakingFor(
...
 	}
 
 	for _, blsKey := range blsKeys {
-		vmOutput, errExec := s.executeOnStakingSC([]byte("unStake@" + hex.EncodeToString(blsKey) + "@" + hex.EncodeToString(registrationData.RewardAddress)))
+		vmOutput, errExec := v.executeOnStakingSC([]byte("unStake@" + hex.EncodeToString(blsKey) + "@" + hex.EncodeToString(registrationData.RewardAddress)))
 		if errExec != nil {
-			s.eei.AddReturnMessage(fmt.Sprintf("cannot do unStake for key %s: %s", hex.EncodeToString(blsKey), errExec.Error()))
-			s.eei.Finish(blsKey)
-			s.eei.Finish([]byte{failed})
+			v.eei.AddReturnMessage(fmt.Sprintf("cannot do unStake for key %v: %v", hex.EncodeToString(blsKey), errExec.Error()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 10]
```diff
@@ -834,185 +824,185 @@ func (s *stakingAuctionSC) activateStakingFor(
...
+			v.eei.Finish(blsKey)
+			v.eei.Finish([]byte{failed})
 			continue
 		}
 
 		if vmOutput.ReturnCode != vmcommon.Ok {
-			s.eei.AddReturnMessage(fmt.Sprintf("cannot do unStake for key %s: %s", hex.EncodeToString(blsKey), vmOutput.ReturnCode.String()))
-			s.eei.Finish(blsKey)
-			s.eei.Finish([]byte{failed})
+			v.eei.AddReturnMessage(fmt.Sprintf("cannot do unStake for key %v: %v", hex.EncodeToString(blsKey), vmOutput.ReturnCode.String()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 11]
```diff
@@ -834,185 +824,185 @@ func (s *stakingAuctionSC) activateStakingFor(
...
 	totalUnBond := big.NewInt(0)
 	totalSlashed := big.NewInt(0)
 	for _, blsKey := range blsKeys {
-		nodeData, errGet := s.getStakedData(blsKey)
+		nodeData, errGet := v.getStakedData(blsKey)
 		if errGet != nil {
-			s.eei.AddReturnMessage(fmt.Sprintf("cannot do unBond for key: %s, error: %s", hex.EncodeToString(blsKey), errGet.Error()))
-			s.eei.Finish(blsKey)
-			s.eei.Finish([]byte{failed})
+			v.eei.AddReturnMessage(fmt.Sprintf("cannot do unBond for key: %v, error: %v", hex.EncodeToString(blsKey), errGet.Error()))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/auction.go [hunk 12]
```diff
@@ -834,185 +824,185 @@ func (s *stakingAuctionSC) activateStakingFor(
...
 			continue
 		}
 		// returns what value is still under the selected bls key
-		vmOutput, errExec := s.executeOnStakingSC([]byte("unBond@" + hex.EncodeToString(blsKey)))
+		vmOutput, errExec := v.executeOnStakingSC([]byte("unBond@" + hex.EncodeToString(blsKey)))
 		if errExec != nil || vmOutput.ReturnCode != vmcommon.Ok {
-			s.eei.AddReturnMessage(fmt.Sprintf("cannot do unBond for key: %s", hex.EncodeToString(blsKey)))
-			s.eei.Finish(blsKey)
-			s.eei.Finish([]byte{failed})
+			v.eei.AddReturnMessage(fmt.Sprintf("cannot do unBond for key: %v", hex.EncodeToString(blsKey)))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction.go]: please revert here

File: vm/systemSmartContracts/proto/validator.proto [hunk 13]
```diff
@@ -36,11 +36,10 @@ message AuctionDataV2 {
...
-    bytes  NodePrice     = 5 [(gogoproto.jsontag) = "NodePrice", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
-    bytes  UnJailPrice   = 6 [(gogoproto.jsontag) = "UnJailPrice", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
+message ValidatorConfig {
+    bytes  MinStakeValue = 1 [(gogoproto.jsontag) = "MinStakeValue", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
+    bytes  TotalSupply   = 2 [(gogoproto.jsontag) = "TotalSupply", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
+    bytes  MinStep       = 3 [(gogoproto.jsontag) = "MinStep", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
+    bytes  NodePrice     = 4 [(gogoproto.jsontag) = "NodePrice", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
+    bytes  UnJailPrice   = 5 [(gogoproto.jsontag) = "UnJailPrice", (gogoproto.casttypewith) = "math/big.Int;github.com/ElrondNetwork/elrond-go/data.BigIntCaster"];
 }
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/proto/validator.proto:45]: empty line

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: vm/systemSmartContracts/auction.go (hunk 3 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:96]: Yes it is. This config was not saved to storage.

File: vm/systemSmartContracts/auction.go (hunk 4 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:689]: done

File: vm/systemSmartContracts/auction.go (hunk 6 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:807]: done

File: vm/systemSmartContracts/auction.go (hunk 5 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:800]: done

File: vm/systemSmartContracts/auction.go (hunk 8 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:841]: done

File: vm/systemSmartContracts/auction.go (hunk 9 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:887]: done

File: vm/systemSmartContracts/auction.go (hunk 10 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:894]: done

File: vm/systemSmartContracts/auction.go (hunk 12 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:997]: done

File: vm/systemSmartContracts/auction.go (hunk 11 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:989]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: vm/systemSmartContracts/auction.go [hunk 14]
```diff
@@ -49,35 +49,31 @@ type stakingAuctionSC struct {
...
 	}
 	if len(args.StakingSCAddress) == 0 {
 		return nil, vm.ErrNilStakingSmartContractAddress
 	}
-	if len(args.AuctionSCAddress) == 0 {
-		return nil, vm.ErrNilAuctionSmartContractAddress
-	}
-	if args.NumOfNodesToSelect < 1 {
-		return nil, fmt.Errorf("%w, value is %v", vm.ErrInvalidMinNumberOfNodes, args.NumOfNodesToSelect)
+	if len(args.ValidatorSCAddress) == 0 {
```
Comment (raduchis) [inline vm/systemSmartContracts/auction.go:75]: is the check just for 0 enough? What happens if the address is not 32 bytes long? Should there also be a check with the MessageSignVerifier internals, should these be always be synchronized in terms of keys and signatures lengths?

File: vm/systemSmartContracts/delegation.go [hunk 15]
```diff
@@ -1303,12 +1303,12 @@ func (d *delegation) claimRewards(args *vmcommon.ContractCallInput) vmcommon.Ret
 	return vmcommon.Ok
 }
 
-func (d *delegation) executeOnAuctionWithValueInArgs(
+func (d *delegation) executeOnValidatorWithValueInArgs(
```
Comment (raduchis) [inline vm/systemSmartContracts/delegation.go:1306]: executeOnValidator**SC**WithValueInArgs?

File: vm/systemSmartContracts/stakingWithWaiting_test.go [hunk 16]
```diff
@@ -170,7 +170,7 @@ func TestStakingWaitingSC_ExecuteStakeStakeUnStakeStakeUnstake(t *testing.T) {
 	stakerPubKey := []byte("blsPubKey")
 
 	blockChainHook := &mock.BlockChainHookStub{}
-	args := createMockArgumentsForAuction()
+	args := createMockArgumentsForValidator()
```
Comment (raduchis) [inline vm/systemSmartContracts/stakingWithWaiting_test.go:173]: createMockArgumentsForValidator**SC**?

File: cmd/node/config/genesisSmartContracts.json [hunk 17]
```diff
@@ -3,7 +3,7 @@
     "owner": "erd1vxy22x0fj4zv6hktmydg8vpfh6euv02cz4yg0aaws6rrad5a5awqgqky80",
     "filename": "./config/genesisContracts/delegation.wasm",
     "vm-type": "0500",
-    "init-parameters": "%auction_sc_address%@03E8@00@030D40@030D40",
+    "init-parameters": "%validator_sc_address%@03E8@00@030D40@030D40",
```
Comment (raduchis) [inline cmd/node/config/genesisSmartContracts.json:6]: is this backwards compatible? does it need changes in the delegation.wasm?

**Review by raduchis — COMMENTED** [review]
no more "auction" in the whole project
minor clarifications needed

File: vm/systemSmartContracts/auction.go (hunk 14 above)
Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:75]: those signatures exist. the input for this arguments are comming from a set of constants, used throughout the project. if one is modified with wrong length a lot of tests will fail.

File: cmd/node/config/genesisSmartContracts.json (hunk 17 above)
Comment (sasurobert) [reply cmd/node/config/genesisSmartContracts.json:6]: no need to change in delegation.wasm. the address is the same.

File: epochStart/metachain/systemSCs.go [hunk 18]
```diff
@@ -648,90 +648,90 @@ func (s *systemSCProcessor) updateOwnersForBlsKeys() error {
...
 	}()
 
-	userAuctionAccount, err := s.getAuctionSystemAccount()
+	userValidatorAccount, err := s.getValidatorSystemAccount()
 	if err != nil {
 		return err
 	}
 
-	auctionAccounts, err := s.getValidAuctionUserAccountsKeys(userAuctionAccount)
+	validatorAccounts, err := s.getValidValidatorUserAccountsKeys(userValidatorAccount)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply epochStart/metachain/systemSCs.go:656]: done

File: integrationTests/multiShard/txScenarios/moveBalance_test.go [hunk 19]
```diff
@@ -55,12 +55,12 @@ func TestTransaction_TransactionMoveBalanceScenarios(t *testing.T) {
 	_ = createAndSendTransaction(nodes[1], players[3], players[3].Address, sendValue, []byte(""), integrationTests.MinTxGasPrice-1, integrationTests.MinTxGasLimit)
 	time.Sleep(100 * time.Millisecond)
 	// send value to staking contract without data should consume gas
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/multiShard/txScenarios/moveBalance_test.go:57]: done

File: integrationTests/multiShard/txScenarios/moveBalance_test.go [hunk 20]
```diff
@@ -55,12 +55,12 @@ func TestTransaction_TransactionMoveBalanceScenarios(t *testing.T) {
...
-	tx3 := createAndSendTransaction(nodes[0], players[4], vm.AuctionSCAddress, sendValue, []byte(""), integrationTests.MinTxGasPrice, integrationTests.MinTxGasLimit)
+	tx3 := createAndSendTransaction(nodes[0], players[4], vm.ValidatorSCAddress, sendValue, []byte(""), integrationTests.MinTxGasPrice, integrationTests.MinTxGasLimit)
 	time.Sleep(100 * time.Millisecond)
 	// send value to staking contract with data should consume gas
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/multiShard/txScenarios/moveBalance_test.go:60]: done

File: vm/factory/systemSCFactory.go [hunk 21]
```diff
@@ -158,21 +158,20 @@ func (scf *systemSCFactory) createStakingContract() (vm.SystemSmartContract, err
...
+		ValidatorSCAddress: vm.ValidatorSCAddress,
 		GasCost:            scf.gasCost,
 		Marshalizer:        scf.marshalizer,
-		NumOfNodesToSelect: scf.systemSCConfig.StakingSystemSCConfig.NodesToSelectInAuction,
 		GenesisTotalSupply: scf.economics.GenesisTotalSupply(),
 		EpochNotifier:      scf.epochNotifier,
 	}
-	auction, err := systemSmartContracts.NewStakingAuctionSmartContract(args)
-	return auction, err
+	validator, err := systemSmartContracts.NewValidatorSmartContract(args)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply vm/factory/systemSCFactory.go:173]: done

File: vm/factory/systemSCFactory.go [hunk 22]
```diff
@@ -252,12 +251,12 @@ func (scf *systemSCFactory) Create() (vm.SystemSCContainer, error) {
 		return nil, err
 	}
 
-	auction, err := scf.createAuctionContract()
+	validator, err := scf.createValidatorContract()
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply vm/factory/systemSCFactory.go:254]: done

File: vm/systemSmartContracts/delegation.go (hunk 15 above)
Comment (sasurobert) [reply vm/systemSmartContracts/delegation.go:1306]: done

File: vm/systemSmartContracts/stakingWithWaiting_test.go (hunk 16 above)
Comment (sasurobert) [reply vm/systemSmartContracts/stakingWithWaiting_test.go]: done

**Review by iulianpascalau — APPROVED** [review]

File: integrationTests/multiShard/txScenarios/moveBalance_test.go (hunk 19 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply integrationTests/multiShard/txScenarios/moveBalance_test.go]: ok

File: epochStart/metachain/systemSCs.go (hunk 18 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply epochStart/metachain/systemSCs.go]: ok

File: integrationTests/multiShard/txScenarios/moveBalance_test.go (hunk 20 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply integrationTests/multiShard/txScenarios/moveBalance_test.go]: ok

File: vm/factory/systemSCFactory.go (hunk 21 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply vm/factory/systemSCFactory.go]: ok

File: vm/factory/systemSCFactory.go (hunk 22 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply vm/factory/systemSCFactory.go]: ok

File: vm/systemSmartContracts/auction.go (hunk 14 above)
Comment (raduchis) [reply vm/systemSmartContracts/auction.go:75]: ok

File: vm/systemSmartContracts/delegation.go (hunk 15 above)
Comment (raduchis) [reply vm/systemSmartContracts/delegation.go]: ok

File: vm/systemSmartContracts/stakingWithWaiting_test.go (hunk 16 above)
Comment (raduchis) [reply vm/systemSmartContracts/stakingWithWaiting_test.go]: ok

File: cmd/node/config/genesisSmartContracts.json (hunk 17 above)
Comment (raduchis) [reply cmd/node/config/genesisSmartContracts.json:6]: ok

**Review by raduchis — APPROVED** [review]
